bundles/org.openhab.binding.visualcrossing/src/main/java/org/openhab/binding/visualcrossing/internal/api/rest/RestClient.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.visualcrossing.internal.api.rest;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.visualcrossing.internal.api.VisualCrossingApiException;
+import org.openhab.binding.visualcrossing.internal.api.VisualCrossingAuthException;
+import org.openhab.binding.visualcrossing.internal.api.VisualCrossingRateException;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public interface RestClient {
+    /**
+     * GET request to server
+     * 
+     * @param url to send request
+     * @param headers to send
+     * @return response from server
+     */
+    @Nullable
+    String get(String url, @Nullable Header... headers)
+            throws VisualCrossingApiException, VisualCrossingAuthException, VisualCrossingRateException;
+
+    /**
+     * POST request to server
+     * 
+     * @param url to send request
+     * @param headers to send
+     * @param content to send
+     * @return response from server
+     */
+    @Nullable
+    String post(String url, Content content, @Nullable Header... headers)
+            throws VisualCrossingApiException, VisualCrossingAuthException, VisualCrossingRateException;
+
+    /**
+     * Represents content with a body and a type.
+     */
+    record Content(String body, String type) {
+        /**
+         * Creates a Content instance with the given body and default type ("application/json").
+         *
+         * @param body The content body.
+         */
+        public Content(String body) {
+            this(body, "application/json");
```

Comment:
Can you sue a mediatype constant like [APPLICATION_JSON ](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/MediaType.html#APPLICATION_JSON)